File: 0429-N-ary-Tree-Level-Order-Traversal/Java/Solution.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class Solution {
    
    /**
     * https://leetcode.cn/problems/n-ary-tree-level-order-traversal/
     * 
     * @param root
     * @return
     */
    public List<List<Integer>> levelOrder(Node root) {
        if (root == null) {
            return new ArrayList<>();
        }

        List<List<Integer>> result = new ArrayList<>();
        Queue<Node> queue = new LinkedList<>();
        queue.add(root);
        while (!queue.isEmpty()) {
            int size = queue.size();
            List<Integer> level = new ArrayList<>();
            for (int i = 0; i < size; i++) {
                Node node = queue.poll();
                level.add(node.val);
                if (node.children != null) {
                    queue.addAll(node.children);
                }
            }
            result.add(level);
        }
        return result;
    }

}

class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};

File: 0429-N-ary-Tree-Level-Order-Traversal/Python/Solution.py
from typing import Optional, List, Deque
from collections import deque


# Definition for a Node.
class Node:
    def __init__(self, val: Optional[int] = None, children: Optional[List['Node']] = None):
        self.val = val
        self.children = children


class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        if root is None:
            return []
        
        result: List[List[int]] = []
        q: Deque[Node] = deque()
        q.append(root)

        while q:
            size: int = len(q)
            sub_list: List[int] = []
            for _i in range(size):
                tn = q.popleft()
                sub_list.append(tn.val)
                for n in tn.children:
                    q.append(n)
            result.append(sub_list)

        return result